src/modules/shared/services/api-service/api.service.ts

```diff
@@ -15,6 +17,16 @@ export abstract class ApiService {
 					'content-type': 'application/json',
 				},
 				credentials: 'include', // TODO change to same-origin once working on server
+				hooks: {
+					afterResponse: [
+						(_request, _options, response) => {
+							if (response.status === 401) {
+								// user is unauthorized and should login again
+								AuthService.logout();
```

Comment:
In this case it might be nice if after the user logs back in, they are redirected back to the page they were trying to visit:
```
	public static logout(shouldRedirectToOriginalPage: boolean): void {
		let returnToUrl = publicRuntimeConfig.CLIENT_URL;
		if (shouldRedirectToOriginalPage) {
			const path = window.location.href.substring(publicRuntimeConfig.CLIENT_URL.length);
			returnToUrl = stringifyUrl({
				url: publicRuntimeConfig.CLIENT_URL,
				query: {
					redirectTo: path,
					showAuth: 1
				}
			});
		}

		window.location.href = stringifyUrl({
			url: `${publicRuntimeConfig.PROXY_URL}/auth/global-logout`,
			query: {
				returnToUrl,
			},
		});
	}
```

Reply:
fixed